src/cluster.c

```diff
@@ -1032,6 +1040,26 @@ clusterNode *clusterLookupNode(const char *name) {
     return dictGetVal(de);
 }
 
+/* Get all the nodes serving this slot. */
+list *clusterGetNodesForSlot(clusterNode *node) {
+    list *nodesForSlot = listCreate();
+    if (node->slaves) {
+        for (int i=0; i < node->numslaves; i++) {
+            listAddNodeTail(nodesForSlot, node->slaves[i]);
+        }
+    } else if (node->slaveof) {
+        clusterNode *master = node->slaveof;
+        listAddNodeTail(nodesForSlot, master);
+        for (int i=0; i < master->numslaves; i++) {
+            if (master->slaves[i] == node) {
+                continue;
+            }
+            listAddNodeTail(nodesForSlot, master->slaves[i]);
+        }
+    }
+    return nodesForSlot;
+}
```

Comment:
This intuitively makes more sense to me.
```suggestion
/* Get all the nodes serving my slos. */
list *clusterGetNodesServingMySlots(clusterNode *node) {
    list *nodesForSlot = listCreate();
    clusterNode *my_primary = nodeIsMaster(myself) ? myself : myself->slaveof;
    listAddNodeTail(nodesForSlot, my_primary);
    for (int i=0; i < my_primary->numslaves; i++) {
        listAddNodeTail(nodesForSlot, my_primary->slaves[i]);
    }
    return nodesForSlot;
}
```

Reply:
Looks cleaner to me.